docs/fast_data/troubleshooting/rtu_violates_unique_keys.md

```diff
@@ -0,0 +1,31 @@
+---
+id: rtu_violates_unique_keys
+title: The Real-Time Updater violates unique keys
+sidebar_label: RTU violates unique keys
+---
+
+## Problem
+
+The Real-Time Updater violates unique keys on MongoDB. This usually happens when a combination of a delete and an insert ingestion message for the same record comes from the CDC, and the collection involved has a secondary unique index that differs from the primary key one.
+
+## Cause
+
+The RTU was executing a compaction of the messages having the same id, resulting in keeping only the insert messages, hence violating the unique key when trying to apply the update on the projections.
```

Comment:
I may be wrong / partially remember, but this wasn't the cause. In fact, the RTU does not perform any compaction. Instead, since there was a duplicate key error, the logic around this particular error (for the _check on offset_ logic) was triggered, but due to the double unique key was failing and kept restarting the service.

Reply:
You are absolutely right! I will update the cause as soon as possible. Thank you!

Reply:
I've now changed the paragraph.